shell/components/Resource/Detail/Card/StateCard/composables.ts

```diff
@@ -0,0 +1,111 @@
+import { extractCounts, Props as ResourceRowProps } from '@shell/components/Resource/Detail/ResourceRow.vue';
+import { useI18n } from '@shell/composables/useI18n';
+import { INGRESS, SERVICE } from '@shell/config/types';
+import { getHighestAlertColor } from '@shell/utils/style';
+import { computed, Ref, toValue } from 'vue';
+import { useStore } from 'vuex';
+import { Props as StateCardProps } from '@shell/components/Resource/Detail/Card/StateCard/index.vue';
+
+export function usesResourceCardRow(label: string, resources: any[], to?: any): ResourceRowProps {
+  const colors = resources.map((r: any) => r.stateSimpleColor);
+  const states = resources.map((r: any) => r.stateDisplay.toLowerCase());
+
+  return {
+    label,
+    color:  resources.length ? getHighestAlertColor(colors) : undefined,
+    counts: resources.length ? extractCounts(states) : undefined,
+    to
+  };
+}
+
+export interface Pairs {
+    label: string;
+    to?: any;
+    resources: any[];
```

Comment:
I think that we should strive to provide more robust types than `any` when possible.